h5glance/ipython.py

```diff
@@ -0,0 +1,33 @@
+import h5py
+
+from .html import h5obj_to_html
+from .terminal import group_to_str
+
+class H5Glance:
+    """View an HDF5 object in a Jupyter notebook"""
+    def __init__(self, obj):
+        self.obj = obj
+
+    def _repr_html_(self):
+        return h5obj_to_html(self.obj)
+
+    def __repr__(self):
+        return group_to_str(self.obj, max_depth=1)
+
+def install_ipython_h5py_display(html=True, text=True):
+    """Call inside IPython to install HTML/text views for h5py groups and files"""
+    from IPython import get_ipython
+    ip = get_ipython()
+    if ip is None:
+        raise EnvironmentError("This function is to be called in IPython")
+
+    if html:
+        html_formatter = ip.display_formatter.formatters['text/html']
+        html_formatter.for_type(h5py.Group, h5obj_to_html)
+    if text:
+        text_formatter = ip.display_formatter.formatters['text/plain']
+        text_formatter.for_type(h5py.Group, pretty_print_group)
+        text_formatter.for_type(h5py.File, pretty_print_group)
+
+def pretty_print_group(obj, p, cycle):
```

Comment:
Why is the "cycle" argument needed?

Reply:
Apart from that, LGTM

Reply:
It's part of the interface IPython expects for pretty-print objects in text; this is an older interface that predates all the newer HTML stuff. It's called here:

https://github.com/ipython/ipython/blob/60625f241f298b5039cb2debc365db38aa7bb522/IPython/lib/pretty.py#L398

The cycle parameter is True when a container is in itself (it's possible to do `a = []; a.append(a)`, although it's not often useful). We're not using it here, but we still need to accept it.

Reply:
Thanks for the explanation!